### PR TITLE
Disables setting AUTOCOMMIT for Postgres >= 9.4.0

### PR DESCRIPTION
**Pull request by tkalfigo**
Fixes issue https://github.com/sequelize/sequelize/issues/4631
User needs to supply "databaseVersion" key in options object
when instantiating the Sequelize class. Value should be semver
compliant e.g. "9.5.0" but not "9.5.0beta2"

### CODE REVIEW

**Comment by mickhansen** [comment]
Version is also auto detected on the first connection if not supplied.

File: lib/dialects/postgres/query-generator.js [hunk 1]
```diff
@@ -873,7 +874,11 @@ var QueryGenerator = {
     }
 
     // POSTGRES does not support setting AUTOCOMMIT = OFF
-    if (!value) {
+    // Additionally it does not support AUTOCOMMIT at all starting at v9.5
+    // The assumption is that it won't be returning in future versions either
+    // The options param passed when creating the Sequelize instance should include a key "databaseVersion" 
+    // with semver compliant value e.g. "9.5.0" (without 'alpha' or 'beta' classifications)
+    if (!value || semver.gte(this.sequelize.options.databaseVersion, '9.4.0')) {
```
Comment (mickhansen) [inline lib/dialects/postgres/query-generator.js]: if starting with 9.5 it needs to be gte 9.5.0 or gt 9.4.0

**Comment by tkalfigo** [comment]
Unfortunately I didn't get this behavior. If I don't supply databaseVersion: "9.5.0" explicitly as part of my options object when instantiating Sequelize, it thinks I'm on "9.4.0".

**Comment by mickhansen** [comment]
https://github.com/sequelize/sequelize/blob/master/lib/dialects/abstract/connection-manager.js#L213
This code path should check version on the first query.

**Comment by mickhansen** [comment]
But if you have a beta version you might be hitting this: https://github.com/sequelize/sequelize/blob/master/lib/dialects/abstract/connection-manager.js#L224 which then defaults to: https://github.com/sequelize/sequelize/blob/master/lib/dialects/postgres/index.js#L51

**Comment by tkalfigo** [comment]
@mickhansen You are right; that explains Sequelize's behavior with 9.5.0beta2.

Regarding the choice to go for gte(9.4.0) and not gte(9.5.0) it's because starting at 9.4.0 AUTOCOMMIT = OFF is *not* allowed; so the only choice is AUTOCOMMIT = ON.

The server itself has autocommit, so every program that uses PostgreSQL will be in autocommit mode by default unless there is an explicit BEGIN or START TRANSACTION. And if you are in a transaction, there is no point doing an AUTOCOMMIT = ON anyway.

Hope I didn't miss any case, but it seems that playing with AUTOCOMMIT through Sequelize is pointless starting at 9.4.0.

**Comment by mickhansen** [comment]
autocommit as a whole is pretty dumb/pointless for transactions, not sure why it ever made it in there.
thanks for the clarification of 9.4.0, could've read the comments more closely :)

**Comment by tkalfigo** [comment]
Is something missing from this to be merged?

**Comment by mickhansen** [comment]
Hmm, i might have been waiting for CI and then forgot about it.
It'll need a rebase now, mind doing that?

**Comment by tkalfigo** [comment]
Done.

**Comment by mickhansen** [comment]
Ah looks like you merged rather than rebases, which messes up history a bit.

**Comment by tkalfigo** [comment]
Will redo PR.

**Pull request closed without merging**